src/core/pathfinding/HPAStar.ts

```
@@ -0,0 +1,117 @@
+        Math.abs(gm.y(tile) - gm.y(gw.pos));
+      if (d < bestD) {
+        bestD = d;
+        best = gw;
+      }
+    }
+    return best;
+  }
+
+  private static cache = new WeakMap<GameMap, AbstractGraph>();
+  private static getOrCreateGraph(gm: GameMap): AbstractGraph {
+    let g = this.cache.get(gm);
```

Biome (1.9.4)

[error] Using this in a static context can be confusing.

this refers to the class.
Unsafe fix: Use the class name instead.


(lint/complexity/noThisInStatic)